view/frontend/web/js/internals/common.js

```diff
@@ -229,8 +232,11 @@ define(['jquery', 'algoliaInstantSearchLib', 'algoliaBase64', 'Magento_PageCache
                     }
 
                 }
-                uiStateProductIndex['sortBy'] = routeParameters.sortBy;
-                uiStateProductIndex['page'] = routeParameters.page;
+                uiStateProductIndex['sortBy'] = algoliaConfig.sortingParameter === 'product_list_order' ?
```

Comment:
Instead of an inline decision can we abstract this into a mixable function that can be invoked by the Search Adapter module?